src/DocExplorer/components/DocExplorer.tsx

```diff
@@ -273,89 +306,112 @@ const parseCurrentUrlHash = (): UrlHashParams => {
     return null;
   }
 
-  if (typeof docType === "string" && !isDocType(docType)) {
+  if (typeof docType === "string" && !isValidDocType(docType)) {
     alert(`Invalid doc type in URL: ${docType}`);
     return null;
   }
 
   return {
-    docUrl,
-    docType,
+    url: docUrl,
+    name: "",
+    type: docType,
   };
 };
 
-// Drive the currently selected doc using the URL hash
+// Drive the currently selected doc using the URL
 // (We encapsulate the selection state in a hook so that the only
 // API for changing the selection is properly thru the URL)
-const useSelectedDoc = ({
+const useSelectedDocLink = ({
   rootFolderDoc,
   changeRootFolderDoc,
 }: {
   rootFolderDoc: FolderDoc;
   changeRootFolderDoc: (fn: (doc: FolderDoc) => void) => void;
-}) => {
-  const [selectedDocUrl, setSelectedDocUrl] = useState<AutomergeUrl>(null);
-  const [selectedDoc] = useDocument(selectedDocUrl);
-
-  const selectDoc = (docUrl: AutomergeUrl | null) => {
-    const doc = rootFolderDoc.docs.find((doc) => doc.url === docUrl);
-    if (!doc) {
-      alert(`Could not find document with URL: ${docUrl}`);
+}): [DocLink, (docLink: DocLink) => void] => {
+  const currentUrl = useCurrentUrl();
+
+  const setSelectedDocLink = (docLink: DocLink) => {
```

Comment:
Just to confirm: does all routing still flow thru setting the URL and then responding to that? (It looks like it does but I'm skimming quickly and can't tell)

I think it's important to preserve that property for a specific reason: it lowers the risk of the URL and displayed doc deviating -- which would be bad for security if I copy-pasted the URL and sent it to someone and it was the wrong doc

Reply:
Yes, the useCurrentUrl intercepts all navigation events and updates the url state